hpx/util/debug/demangle_helper.hpp

```diff
@@ -0,0 +1,126 @@
+//  Copyright (c) 2017 John Biddiscombe
+//  Copyright (c) 2007-2012 Hartmut Kaiser
+//
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#if !defined(HPX_UTIL_DEBUG_DEMANGLE_HELPER_OCT_28_2011_0410PM)
+#define HPX_UTIL_DEBUG_DEMANGLE_HELPER_OCT_28_2011_0410PM
+
+#include <hpx/config.hpp>
+#include <string>
+#include <type_traits>
+#include <typeinfo>
+
+// --------------------------------------------------------------------
+// Always present regardless of compiler : used by serialization code
+// --------------------------------------------------------------------
+namespace hpx { namespace util { namespace debug
+{
+    template <typename T>
+    struct demangle_helper
+    {
+        char const* type_id() const
+        {
+            return typeid(T).name();
+        }
+    };
+}}}
+
+#if defined(__GNUG__)
+
+#include <cxxabi.h>
+#include <stdlib.h>
+
+// --------------------------------------------------------------------
+// if available : demangle an arbitrary c++ type using gnu utility
+// --------------------------------------------------------------------
+namespace hpx { namespace util { namespace debug
+{
+    template <typename T>
+    class cxxabi_demangle_helper
+    {
+    public:
+        cxxabi_demangle_helper()
+          : demangled_(abi::__cxa_demangle(typeid(T).name(), 0, 0, 0))
+        {}
+
+        ~cxxabi_demangle_helper()
+        {
+            free(demangled_);
```

Comment:
Due to this, the class can't be copied. I think it would make sense to use proper memory management here. As I suggested previously, using either `std::vector<char>`, `std::unique_ptr<char, deleter>` or even `std::string` might help here.

Reply:
just to clarify - the code in question was copied directly from the old implementation and used only in one place to initialize a static var that is used for the symbol translation (so copying is not planned or supported for this class). I will however use a unique_ptr if that makes you happy.